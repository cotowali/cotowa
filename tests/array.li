fn echo_arr(arr []int) {
  echo('echo_arr')
  arr
}

fn raw_arr(arr []int) []int {
  return arr
}

fn new_arr() []int {
  return [0, 1, 2]
}

var n = 2
var a = 'a'

[0, 1, n]
var v = [a, 'b', 'c']
v
v = []string{}

var arr = [0, 1, n]
echo_arr(arr)
echo_arr([0, 1, 2])
raw_arr(arr)
new_arr()
// assert raw_arr(arr) == arr

fn for_in_test() {
  echo('for')
  for i in [0, 1, 2] {
    echo(i)
  }
}
for_in_test()

fn index_test() {
  var arr = [10, 11, 12]
  assert arr[0] == 10
  assert arr[1] == 11
  arr[2] = 100
  assert arr[2] == 100

  arr = [10, 11, 12]
  var i = 0
  assert arr[i] == 10
  i = 2
  assert arr[i] == 12
  arr[i] = 0
  assert arr[i] == 0
}
index_test()
