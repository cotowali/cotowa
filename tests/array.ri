fn echo_arr(arr []int) {
  echo('echo_arr')
  arr
}

fn raw_arr(arr []int) []int {
  return arr
}

[0, 1, 2]
let v = ['a', 'b', 'c']
v

let arr = [0, 1, 2]
echo_arr(arr)
echo_arr([0, 1, 2])
raw_arr(arr)
// assert raw_arr(arr) == arr
