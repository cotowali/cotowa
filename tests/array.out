0 1 2
a b c
echo_arr
0 1 2
echo_arr
0 1 2
0 1 2
